input/signdescriptor.go

```diff
@@ -74,6 +74,11 @@ type SignDescriptor struct {
 	// generating the final sighash for signing.
 	SigHashes *txscript.TxSigHashes
 
+	// PrevOutputFetcher is an interface that can return the output
+	// information on all UTXOs that are being spent in this transaction.
+	// This MUST be set when spending Taproot outputs.
+	PrevOutputFetcher txscript.PrevOutputFetcher
```

Comment:
`txscript.PrevOutputFetcher` is an interface, so it's basically a pointer. Or you mean a pointer to the interface to avoid that nil equality check against interfaces?